lib/dao.js

```diff
@@ -1562,7 +1565,23 @@ DataAccessObject._coerce = function(where) {
 
       continue;
     }
+
     var DataType = props[p] && props[p].type;
+    if (p.match(/\./)) {
+      var model = p.split('.')[0];
+      var prop = p.split('.').slice(1);
+
+      if (props[model]) {
+        var clause = {};
+        clause[prop] = where[p];
+        where[p] = Array.isArray(props[model].type) ?
+          self._coerce(clause, props[model].type[0].definition.properties)[prop] :
```

Comment:
That test passes with the current implementation. I need a test that is failing now and passes with your changes in place.

The test you are referring to is querying a string value, so there is no need to perform coercion. We need to query a date value instead.